pex/variables.py

```diff
@@ -210,15 +210,40 @@ def _get_kv(cls, variable):
         if len(list(filter(None, kv))) == 2:
             return kv
 
+    @staticmethod
+    def _maybe_get_bool_var(
+        name,  # type: str
+        env,  # type: Mapping[str, Optional[str]]
+    ):
+        # type: (...) -> Optional[bool]
+        value = env.get(name, None)
+        if value is None:
+            return None
+        if value.lower() in ("0", "false"):
+            return False
+        if value.lower() in ("1", "true"):
+            return True
+        raise ValueError(
+            "Invalid bool value for {name}, must be 0/1/false/true, got {value!r}".format(
+                name=name, value=value
+            )
+        )
+
     def __init__(self, environ=None, rc=None):
-        # type: (Optional[Dict[str, str]], Optional[str]) -> None
-        self._environ = (
-            environ.copy() if environ is not None else os.environ.copy()
-        )  # type: Dict[str, str]
-        if not self.PEX_IGNORE_RCFILES:
-            rc_values = self.from_rc(rc).copy()
-            rc_values.update(self._environ)
-            self._environ = rc_values
+        # type: (Optional[Mapping[str, str]], Optional[str]) -> None
+        env = environ if environ is not None else os.environ
+        if self._maybe_get_bool_var("PEX_DISABLE_VARIABLES", env) is True:
+            self._environ = {
+                key: value
+                for key, value in env.items()
+                if key == "PEX_DISABLE_VARIABLES" or not key.startswith("PEX")
```

Comment:
```suggestion
                if key == "PEX_DISABLE_VARIABLES" or not key.startswith("PEX_")
```
To match the documentation below? Not that there would be a difference in practice I suppose.

Reply:
Fixed.